app/server.go

```diff
@@ -727,11 +728,32 @@ func runSecurityJob(s *Server) {
 	}, time.Hour*4)
 }
 
+func doDiagnosticsIfNeeded(s *Server, firstRun time.Time) {
```

Comment:
Given the importance of correctly running diagnostics, I would really like to see tests for the change.

Reply:
@hanzei , will add tests.

Reply:
We have tests for the diagnostics feature generally. I don't think it's necessary to insist on tests for the new timing calculation - a manual test should be sufficient here. Once we clean up and remove the Segment stuff for 5.24 or 5.26, we could look at a more general refactor to make more of this feature testable through automated tests.